framework/src/main/java/org/checkerframework/framework/type/ElementAnnotationApplier.java

```diff
@@ -138,7 +167,13 @@ private static void applyInternal(
      */
     public static void annotateSupers(
             List<AnnotatedDeclaredType> supertypes, TypeElement subtypeElement) {
-        SuperTypeApplier.annotateSupers(supertypes, subtypeElement);
+        try {
+            SuperTypeApplier.annotateSupers(supertypes, subtypeElement);
+        } catch (UnexpectedAnnotationLocationException e) {
+            if (!supertypes.isEmpty()) {
```

Comment:
Could you add a comment why this guard is here? If `supertypes` were empty, how would the exception ever be raised?